src/cdi_ndi/NDI_AMW.cc

```diff
@@ -0,0 +1,81 @@
+//----------------------------------*-C++-*-----------------------------------//
+/*!
+ * \file   cdi_ndi/NDI_AMW.cc
+ * \author Ben R. Ryan
+ * \date   2020 Mar 6
+ * \brief  NDI_AMW class declaration.
+ * \note   Copyright (C) 2020 Triad National Security, LLC.
+ *         All rights reserved. */
+//----------------------------------------------------------------------------//
+
+#include "NDI_AMW.hh"
+#include "ndi.h"
+#include "cdi_ndi/config.h"
+#include "ds++/Assert.hh"
+
+namespace rtt_cdi_ndi {
+
+//----------------------------------------------------------------------------//
+// CONSTRUCTORS
+//----------------------------------------------------------------------------//
+/*!
+ * \brief Constructor for NDI atomic mass weight reader, using default path to
+ *        NDI gendir file.
+ */
+NDI_AMW::NDI_AMW() {
+  gendir_path = std::string(NDI_ROOT_DIR) + "/share/gendir.all";
```

Comment:
1. This should be in an initializer list for the constructor.  
2. This path will be invalid on Windows platforms, so you will need to use ds++'s `path::getFilenameComponent` with the `TO_NATIVE` argument to normalize the path.
3. Please add a DbC in the ctor body to check that the file exists (see `fileExists` function from ds++/path.hh).